coral/src/app/features/approvals/acls/components/AclApprovalsTable.tsx

```diff
@@ -0,0 +1,311 @@
+              />
+            ))}
+          </Flexbox>
+        );
+      },
+    },
+    {
+      type: "custom",
+      field: "topicname",
+      headerName: "Topic",
+      UNSAFE_render({ topicname, prefixed }: AclRequestTableRow) {
+        return (
+          <>
+            {topicname}
+            {prefixed && <code>(prefixed)</code>}
+          </>
+        );
+      },
+    },
+    {
+      type: "status",
+      field: "environmentName",
+      headerName: "Environment",
+      status: ({ environmentName }) => ({
+        status: "neutral",
+        text: environmentName,
+      }),
+    },
+    {
+      type: "text",
+      field: "teamname",
+      headerName: "Team",
+    },
+    {
+      type: "status",
+      field: "aclType",
+      headerName: "ACL type",
+      status: ({ aclType }) => ({
+        status: aclType === "CONSUMER" ? "success" : "info",
+        text: aclType,
+      }),
+    },
+    {
+      type: "status",
+      field: "requestStatus",
+      headerName: "Status",
+      status: ({ requestStatus }) => {
+        if (requestStatus === "") {
+          return {
+            status: "neutral",
+            text: "-",
+          };
+        }
+        return {
+          status: requestStatusChipStatusMap[requestStatus],
+          text: requestStatusNameMap[requestStatus],
+        };
+      },
+    },
+    {
+      type: "status",
+      field: "requestOperationType",
+      headerName: "Request type",
+      status: ({ requestOperationType }) => {
+        if (requestOperationType === "") {
+          return {
+            status: "neutral",
+            text: "-",
+          };
+        }
+        return {
+          status: requestOperationTypeChipStatusMap[requestOperationType],
+          text: requestOperationTypeNameMap[requestOperationType],
+        };
+      },
+    },
+    {
+      type: "text",
+      field: "username",
+      headerName: "Requested by",
+    },
+    {
+      type: "text",
+      field: "requesttimestring",
+      headerName: "Requested on",
+      formatter: (value) => {
+        return `${value}${"\u00A0"}UTC`;
+      },
+    },
+    {
+      headerName: "Details",
+      headerInvisible: true,
+      type: "custom",
+      UNSAFE_render: ({ id, topicname }: AclRequestTableRow) => {
+        return (
+          <GhostButton
+            icon={infoSign}
+            aria-label={`View acl request for ${topicname}`}
+            onClick={() => onDetails(String(id))}
+            dense
+          >
+            <span aria-hidden={"true"}>View details</span>
+            <span className={"visually-hidden"}>
+              View topic request for {topicname}
+            </span>
+          </GhostButton>
+        );
+      },
+    },
+    {
+      width: 30,
+      headerName: "Approve",
+      headerInvisible: true,
+      type: "custom",
+      UNSAFE_render: ({ id, requestStatus, topicname }: AclRequestTableRow) => {
+        const [isLoading, setIsLoading] = useState(false);
```

Comment:
I believe that this local state may be unnecessary now, thanks to getting the `variables` from the `useMutation`, and the implementation of the `isBeingApproved`   function.

Reply:
Yes, let's drop this.